File: src/frontend/test/src/domain/entities/PrestationProposee.ts
import type { ILogement } from '@/domain/entities/Logement';
import type { IPrestation } from '@/domain/entities/Prestation';

export interface IPrestationProposee {
    idPrestationProposee: number;
    prixPrestation: number;
    logement?: ILogement;
    prestation?: IPrestation;
}

export class PrestationProposee implements IPrestationProposee {
    constructor(
        public idPrestationProposee: number,
        public prixPrestation: number,
        public logement?: ILogement,
        public prestation?: IPrestation
    ) {}
}

File: src/backend/NHP_Domain/Entities/PrestationProposee.cs
namespace NHP_Domain.Entities;

public class PrestationProposee
{
    public long IdPrestationProposee { get; set; }
    public long PrixPrestation { get; set; }
    
    //ManyToOne
    public virtual Logement? Logement { get; set; }
    public virtual Prestation? Prestation { get; set; }

    public override string ToString()
    {
        return $"nomLogement: {Logement} " +
               $"prixPrestation: {PrixPrestation} " +
               $"nomPrestation: {Prestation}" +
               $"idPrestationProposee: {IdPrestationProposee}";
    }
}